torchao/prototype/blockwise_fp8_training/scaled_mm_kernels.py

```diff
@@ -0,0 +1,217 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD 3-Clause license found in the
+# LICENSE file in the root directory of this source tree.
+
+"""
+Implementation of blockwise fp8 GEMM operations using torch._scaled_mm
+as an alternative to custom Triton kernels.
+
+This implementation uses block-by-block processing with torch._scaled_mm to maintain
+blockwise scaling precision, providing accurate results comparable to the Triton kernels.
+While torch._scaled_mm doesn't natively support arbitrary blockwise scaling, the 
```

Comment:
`torch._scaled_mm` does natively support blockwise scaling, see [here](https://github.com/pytorch/pytorch/blob/371eacb2ae4ecdabc52ea4634ed21558df2f3bab/aten/src/ATen/cuda/CUDABlas.h#L144-L145). You just need to upgrade to CUDA 12.9 to use it :)

That is the goal of this issue, to support torch._scaled_mm with native blockwise scaled GEMMs (not emulated ones like this).

Reply:
This unit test should help you figure out how to use the torch._scaled_mm native blockwise scaled gemm: https://github.com/pytorch/pytorch/blob/371eacb2ae4ecdabc52ea4634ed21558df2f3bab/test/test_matmul_cuda.py#L1373